src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -2,23 +2,207 @@
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using System.Linq.Expressions;
 using System.Text.RegularExpressions;
 using GeneratorV2.Data;
 using GeneratorV2.Writing;
 
 namespace GeneratorV2.Process
 {
+    public sealed class VectorOverloader : IOverloader
+    {
+        private static readonly Regex VectorNameMatch = new Regex("([1-4])([f|d|h|i])v", RegexOptions.Compiled);
+        private readonly HashSet<string> _existingVectorTypes = new HashSet<string>()
+        {
+            "Vector2",
+            "Vector2d",
+            "Vector2h",
+            "Vector2i",
+            "Vector3",
+            "Vector3d",
+            "Vector3h",
+            "Vector3i",
+            "Vector4",
+            "Vector4d",
+            "Vector4h",
+            "Vector4i",
+        };
+
+        public bool TryGenerateOverloads(Overload overload, [NotNullWhen(true)] out List<Overload>? newOverloads)
+        {
+            Match m = VectorNameMatch.Match(overload.OverloadName);
+            if (!m.Success || !int.TryParse(m.Groups[1].Value, out int vectorSize))
+            {
+                newOverloads = null;
+                return false;
+            }
+            var vectorType = m.Groups[2].Value;
+            if (vectorType == "f") vectorType = "";
+
+            NameTable nameTable = overload.NameTable.New();
+            Parameter[] singleParameters = overload.InputParameters.ToArray();
+            Parameter[] spanParameters = overload.InputParameters.ToArray();
+            Parameter[] arrayParameters = overload.InputParameters.ToArray();
+            List<Parameter> countParameters = new();
+            List<(Parameter ptrParam, Parameter vectorParam, string firstElem)> overloadedParams = new();
+
+            for (var i = 0; i < singleParameters.Length; i++)
+            {
+                Parameter parameter = singleParameters[i];
+                if (parameter.Length == null ||
+                    parameter.Type is not CSPointer ptr || ptr.BaseType is not CSType baseType)
+                {
+                    continue;
+                }
```

Comment:
This if-statement is pretty hard to read and reason about. Negative logic like this is pretty uncommon to read so it becomes unnecessarily hard to understand.
I think we should go for the easier for read version where we just check for the conditions and move the code after into the if-statement. Personally I also find it more intuitive that way.